### PR TITLE
Generate Azure.ResourceManager.CommonTypes v6

### PR DESCRIPTION
**Pull request by AlitzelMendez**
There are 3 common types that changed on this version:  https://github.com/Azure/azure-rest-api-specs/tree/main/specification/common-types/resource-management/v6
- [managedidentity.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/managedidentity.json)
- [privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/privatelinks.json)
- [types.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v6/types.json)

issue: https://github.com/Azure/typespec-azure/issues/1028

### CODE REVIEW

**Comment by AlitzelMendez** [comment]
@allenjzhang, I started this draft pull request with the v6 common types, please lets talk about the common types that doesn't exist yet

**Comment by AlitzelMendez** [comment]
> @allenjzhang, I started this draft pull request with the v6 common types, please lets talk about the common types that doesn't exist yet

I talked with Allen about this offline, he mentioned that for the common-types that are not present of v6 we shouldn't do any change but still produce the types

**Review by markcowl — CHANGES_REQUESTED** [review]

File: packages/typespec-azure-resource-manager/lib/common-types/types.tsp [hunk 1]
```diff
@@ -334,13 +336,16 @@ union ResourceIdentityType {
...
+@removed(Versions.v6)
 model Identity {
   /** The principal ID of resource identity. The value must be an UUID. */
+  @removed(Versions.v6)
   @typeChangedFrom(Versions.v4, string)
   @visibility("read")
   principalId?: uuid;
 
   /** The tenant ID of resource. The value must be an UUID. */
+  @removed(Versions.v6)
```
[the remark this answers is not in the document]
Comment (AlitzelMendez) [reply packages/typespec-azure-resource-manager/lib/common-types/types.tsp:348]: I am getting the following error when only removing to the top level:
 
`'Azure.ResourceManager.CommonTypes.Identity' was removed in version 'v6' but contains type 'Azure.ResourceManager.CommonTypes.Identity.principalId' removed in version ''.TypeSpec(@typespec/versioning/incompatible-versioned-reference)`

looks like a conflict for the `@typeChangedFrom` is this a bug?

[the remark this answers is not in the document]
Comment (allenjzhang) [reply packages/typespec-azure-resource-manager/lib/common-types/types.tsp:348]: Yeah. It is a bug.  Opend https://github.com/microsoft/typespec/issues/4964

**Review by allenjzhang — APPROVED** [review]

**Comment by allenjzhang** [comment]
Updating -ref files is a lot of hassle. We need to look for a way to simplify/eliminate the process. Opened #1800 to track this.

**Comment by AlitzelMendez** [comment]
validation pull requests:
- https://github.com/Azure/azure-rest-api-specs-pr/pull/20274
- https://github.com/Azure/azure-rest-api-specs/pull/31461

**Review by markcowl — APPROVED** [review]

**Comment by abatishchev** [comment]
hi @AlitzelMendez, great news, thank you for working on this! I wonder when I'll be able to consume this change and how it would look like (my TypeSpec file wise).

**Pull request merged**